plugins/action_generate_review_assignment_records/src/generateReviewAssignments.test.ts

```diff
@@ -31,184 +32,232 @@ test('Test: Submit Application ID#4001 - Stage 1 (Last level)', () => {
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-          {
-            reviewerId: 3,
-            organisationId: null,
-            stageId: 5,
-            stageNumber: 1,
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-          {
-            reviewerId: 4,
-            organisationId: null,
-            stageId: 5,
-            stageNumber: 1,
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-          {
-            reviewerId: 5,
-            organisationId: null,
-            stageId: 5,
-            stageNumber: 1,
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-          {
-            reviewerId: 7,
-            organisationId: null,
-            stageId: 5,
-            stageNumber: 1,
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-          {
-            reviewerId: 8,
-            organisationId: null,
-            stageId: 5,
-            stageNumber: 1,
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-          {
-            reviewerId: 9,
-            organisationId: null,
-            stageId: 5,
-            stageNumber: 1,
-            status: ReviewAssignmentStatus.AvailableForSelfAssignment,
-            applicationId: 4001,
-            allowedSections: null,
-            levelNumber: 1,
-            isLastLevel: true,
-            isLastStage: false,
-            isFinalDecision: false,
-          },
-        ],
-        reviewAssignmentIds: [1, 2, 3, 4, 1005, 6, 7],
-        reviewAssignmentAssignerJoins: [],
-        reviewAssignmentAssignerJoinIds: [],
-        nextStageNumber: 1,
```

Comment:
As pointed out by @CarlosNZ the reason this assignments are getting created/updated is because when we first manually generated reviewAssignments for applicationId 4001, only 1 was created, but there are 7 users with proper permission. So, 6 new are being generated and 1 is updated (reviewAssignment 1005, with new status: `AvailableForSelfAssignment`. 
![Screen Shot 2021-08-10 at 12 35 34 PM](https://user-images.githubusercontent.com/16461988/128791272-0570bf6d-e75f-4792-9336-0c00b2a73917.png)

![Screen Shot 2021-08-10 at 12 35 44 PM](https://user-images.githubusercontent.com/16461988/128791276-86f87c30-c3a3-40ef-afb7-149c2aa2b226.png)

The proper fix for this issue will be to prevent new reviewAssignments generated to have this status if one is already **Assigned**. A new PR will be created to proper fix this. And this one can be closed - without merging.